albatross/covariance_functions/covariance_functions.h

```diff
@@ -24,6 +24,10 @@ template <typename Term>
 struct CovarianceFunction {
   Term covariance_term;
 
+  CovarianceFunction() : covariance_term() {};
+  CovarianceFunction(Term &term) : covariance_term(term) {};
+  CovarianceFunction(Term &&term) : covariance_term(term) {};
```

Comment:
Do you need both of these? I think defining the & version should get you the && version via implicit casting. I could be wrong though.

Reply:
yep, you're right.  Thought to get it to work I had to make the `&` version `const`.